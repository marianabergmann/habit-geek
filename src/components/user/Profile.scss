.profile {
  width: 100%;
  max-width: 340px;
  padding: 15px;
  margin: auto;
}

.profile__title {
  text-align: center;
}

.profile__avatar-container {
  display: flex;
  justify-content: space-around;
  height: 50px;
}

.profile__avatar {
  width: 40px;
  border-radius: 50%;
}

.profile__avatar--selected {
  transition-property: width;
  transition-duration: 0.2s;
  border: 3px solid var(--primary);
  outline: none;
  width: 50px;
}
