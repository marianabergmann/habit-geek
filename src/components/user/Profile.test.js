import React from "react";
import { navigate } from "@reach/router";

import { shallow } from "enzyme";
import Profile from "./Profile";

jest.mock("@reach/router", () => ({
  navigate: jest.fn()
}));

afterAll(() => {
  jest.unmock("@reach/router");
});

describe("Profile component", () => {
  it("Renders static elements", () => {
    const profile = shallow(<Profile avatar="002-woman.svg" name="rafa" />);
    expect(profile).toMatchSnapshot();
  });

  describe("on register", () => {
    it("Register and navigates to Dashboard", () => {
      const registerUserAction = jest.fn();
      const profile = shallow(<Profile registerUser={registerUserAction} />);
      profile
        .find("#profile-name")
        .simulate("change", { target: { value: "Rafa" } });
      profile
        .find(".profile__avatar")
        .at(0)
        .simulate("click");
      profile.find("button").simulate("click");

      expect(registerUserAction).toHaveBeenCalled();
      expect(registerUserAction).toHaveBeenCalledWith(
        "Rafa",
        "001-burglar.svg"
      );
      expect(navigate).toHaveBeenCalledWith("/habits");
    });
  });
});
