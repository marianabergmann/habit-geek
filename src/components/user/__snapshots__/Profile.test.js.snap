// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component Renders static elements 1`] = `
<div
  className="mobile-size-container"
>
  <h2>
    Super hero who?
  </h2>
  <div
    className="mb-3"
  >
    <label
      htmlFor="profile-name"
    >
      Name:
    </label>
    <input
      className="form-control"
      id="profile-name"
      onChange={[Function]}
      value="rafa"
    />
  </div>
  <div
    className="mb-3"
  >
    <label
      htmlFor="profile-avatar"
    >
      Avatar:
    </label>
    <div
      className="input-group profile__avatar-container"
    >
      <input
        alt="Search"
        className="profile__avatar "
        key="avatar-0"
        onClick={[Function]}
        src="001-burglar.svg"
        type="image"
      />
      <input
        alt="Search"
        className="profile__avatar profile__avatar--selected"
        key="avatar-1"
        onClick={[Function]}
        src="002-woman.svg"
        type="image"
      />
      <input
        alt="Search"
        className="profile__avatar "
        key="avatar-2"
        onClick={[Function]}
        src="003-superhero.svg"
        type="image"
      />
      <input
        alt="Search"
        className="profile__avatar "
        key="avatar-3"
        onClick={[Function]}
        src="004-robot.svg"
        type="image"
      />
      <input
        alt="Search"
        className="profile__avatar "
        key="avatar-4"
        onClick={[Function]}
        src="005-dragon.svg"
        type="image"
      />
      <input
        alt="Search"
        className="profile__avatar "
        key="avatar-5"
        onClick={[Function]}
        src="006-cyborg.svg"
        type="image"
      />
    </div>
  </div>
  <hr
    className="mb-4"
  />
  <button
    className="btn btn-primary btn-lg btn-block"
    onClick={[Function]}
  >
    Done!
  </button>
</div>
`;
